halo2_proofs/src/plonk/circuit.rs

```diff
@@ -334,6 +335,68 @@ impl TableColumn {
     }
 }
 
+/// Users of halo2 cannot use this struct.
+/// The index of a dynamic table in `ConstraintSystem.dynamic_tables`.
+/// The `index + 1` serves as this dynamic table's unique tag value.
+#[derive(Clone, Copy, Debug, Eq, PartialEq, PartialOrd, Ord, Hash)]
+pub struct DynamicTable(usize);
+
+impl DynamicTable {
+    pub(crate) fn index(self) -> usize {
+        self.0
+    }
```

Comment:
This can be const i think.
```suggestion
    pub(crate) const fn index(self) -> usize {
        self.0
    }
```